.github/workflows/smoke-test/smoke-test.sh

```diff
@@ -0,0 +1,231 @@
+    cleanups+=("hdiutil detach '$mountpoint'")
+
+    codesign --verify --deep --strict --verbose=2 --check-notarization "$srcApp"
+    mkdir -p "$destApp"
+    cleanups+=("rm -rf '$destApp'")
+
+    cp -a "$srcApp" "$(dirname "$destApp")"
+    xattr -d -r -s -v com.apple.quarantine "$destApp"
+
+    if [[ "$(uname -m)" =~ arm ]]; then
+        # For macOS, currently only x86_64 runners support nested virtualization
+        # https://github.com/actions/runner-images/issues/9460
+        # Abort the script (gracefully) instead of trying to run RD.
+        exit 0
+    fi
+
+    RDCTL="$destApp/Contents/Resources/resources/darwin/bin/rdctl"
+}
+
+# Assume the first argument given is a path to the Rancher Desktop zip file;
+# install it, and set the global variable RDCTL to the path of the rdctl
+# executable.
+install_linux() {
+    local archiveName=$1
+
+    sudo mkdir -p /opt/rancher-desktop
+    sudo unzip -d /opt/rancher-desktop "$archiveName"
+
+    RDCTL="/opt/rancher-desktop/resources/resources/linux/bin/rdctl"
+}
+
+# Helper function on Windows to verify the signature of a file (provided as the
+# first argument).
+win32_verify() {
+    local path
+    path="$(cygpath --windows "$1")"
+    # When running GitHub actions, using `powershell.exe` here causes issues
+    # with loading the `Microsoft.PowerShell.Security` module; using `pwsh.exe`
+    # seems to be fine.  This is probably because the default shell is pwsh, and
+    # the environment has paths to the PowerShell 7 version of the module, so it
+    # tries to load that instead of the version appropriate for PowerShell.exe.
+    local pwsh=(pwsh.exe -NoLogo -NoProfile -NonInteractive -Command)
+    local stdout
+    stdout=$("${pwsh[@]}" "\$(Get-AuthenticodeSignature '$path').Status")
+
+    if [[ "$stdout" != "Valid" ]]; then
+        printf "%s is not correctly signed:\n" "$path"
+        "${pwsh[@]}" "Get-AuthenticodeSignature '$path' | Format-List"
+        exit 1
+    fi
+}
+
+# Assume the first argument given is a path to the Rancher Desktop installer;
+# install it, and set the global variable RDCTL to the path of the rdctl
+# executable.
+install_win32() {
+    local archiveName=$1
+
+    win32_verify "$archiveName"
+    mkdir -p "$(cygpath --unix "${RD_LOGS_DIR}")"
+    msiexec.exe '/lv*x' "${RD_LOGS_DIR}\\install.log" \
+        /i "$(cygpath --windows "$archiveName")" /passive ALLUSERS=1
+    # msiexec returns immediately and runs in the background; wait for that
+    # process to exit before continuing.
+    local deadline completed
+    deadline=$(( $(date +%s) + 10 * 60 ))
+    while [[ $(date +%s) -lt $deadline ]]; do
+        if tasklist.exe /FI "ImageName eq msiexec.exe" | grep msiexec; then
+            printf "Waiting for msiexec to finish: %s/%s\n" "$(date)" "$(date --date="@$deadline")"
+            sleep 10
+        else
+            completed=true
+            break
+        fi
+    done
+    if [[ -z "$completed" ]]; then
```

Comment:
Doesn't this fail because of `set -o nounset`?

```suggestion
    if [[ -z "${completed:-}" ]]; then
```

Reply:
No, because it's declared on line 146.

Reply:
It is declared, but not bound to a value. Declaring doesn't set it to the empty string:

```bash
#!/usr/bin/env bash

set -o errexit -o nounset

foo() {
    local completed
    if [[ -z "$completed" ]]; then
        echo not completed
    fi
    echo foo
}

foo
```

Fails:

```console
$ ./nounset.sh
./nounset.sh: line 7: completed: unbound variable
```

You would have to declare it like `local completed=""` to make it work if you don't want to change the condition to `[[ -z "${completed:-}" ]]`.

Reply:
Ugh. I tested this before replying, but I specifically tested the version shipped with macOS… the newer version changes the behaviour.

This works on `3.2.57(1)-release`
But it fails on `5.2.15(1)-release`.

I'll go fix it.

Reply:
Done.